platform/security/servlet/security-servlet-logout/src/test/java/org/codice/ddf/security/servlet/logout/TestLogoutServlet.java

```diff
@@ -33,40 +33,81 @@
 import javax.servlet.http.HttpSession;
 import org.apache.shiro.subject.Subject;
 import org.apache.shiro.util.ThreadContext;
+import org.junit.Before;
 import org.junit.Test;
 
 public class TestLogoutServlet {
-  @Test
-  public void testLocalLogout() {
+  LocalLogoutServlet localLogoutServlet;
+  HttpServletRequest request;
+  HttpServletResponse response;
+  HttpSession httpSession;
+
+  @Before
+  public void testsetup() {
+    localLogoutServlet = new MockLocalLogoutServlet();
+    request = mock(HttpServletRequest.class);
+    response = mock(HttpServletResponse.class);
+
+    httpSession = mock(HttpSession.class);
+    when(request.getSession()).thenReturn(httpSession);
+    when(request.getSession().getId()).thenReturn("id");
+    when(request.getRequestURL()).thenReturn(new StringBuffer("http://foo.bar"));
 
-    LocalLogoutServlet localLogoutServlet = new MockLocalLogoutServlet();
-    HttpServletRequest request = mock(HttpServletRequest.class);
-    HttpServletResponse response = mock(HttpServletResponse.class);
     Subject subject = mock(Subject.class);
     when(subject.hasRole(anyString())).thenReturn(false);
     ThreadContext.bind(subject);
 
     System.setProperty("security.audit.roles", "none");
+  }
 
-    HttpSession httpSession = mock(HttpSession.class);
-    when(request.getSession()).thenReturn(httpSession);
-    when(request.getSession().getId()).thenReturn("id");
-    when(request.getRequestURL()).thenReturn(new StringBuffer("http://foo.bar"));
+  @Test
+  public void testLocalLogout() {
+    // Used for detecting basic auth
+    when(request.getHeaders(anyString())).thenReturn(new LogoutServletEnumeration());
+
+    // used for detecting pki
+    when(request.getAttribute("javax.servlet.request.X509Certificate"))
+        .thenReturn(new X509Certificate[] {mock(X509Certificate.class)});
+
+    SecurityTokenHolder securityTokenHolder = mock(SecurityTokenHolder.class);
+    when(httpSession.getAttribute(SecurityConstants.SAML_ASSERTION))
+        .thenReturn(securityTokenHolder);
+    try {
+      localLogoutServlet.doGet(request, response);
+    } catch (ServletException | IOException e) {
+      fail(e.getMessage());
+    }
+    verify(httpSession).invalidate();
+  }
+
+  @Test()
+  public void testNullSubject() throws Exception {
+    ThreadContext.bind((Subject) null);
+
+    // Used for detecting basic auth
+    when(request.getHeaders(anyString())).thenReturn(new LogoutServletEnumeration());
+
+    // used for detecting pki
+    when(request.getAttribute("javax.servlet.request.X509Certificate"))
+        .thenReturn(new X509Certificate[] {mock(X509Certificate.class)});
+
+    SecurityTokenHolder securityTokenHolder = mock(SecurityTokenHolder.class);
+    when(httpSession.getAttribute(SecurityConstants.SAML_ASSERTION))
+        .thenReturn(securityTokenHolder);
+    try {
```

Comment:
Just curious why we are catching the exception and calling fail instead of just letting the exception fail the test.

Reply:
Good question indeed. I am just reusing the existing unit test. It doesn't seem like it would do any harm to remove it... So if that is the consensus, I can make that change. I think it was originally @stustison 's code, maybe he will have an idea on why or give the thumbs up to change.